### DISCUSSION TITLE
Modify artist/album/track before searching Spotify?

### DISCUSSION THREAD

**Discussion by inderpreet99** [beetbox/beets#4208] [Q&A]
Hi,

I have a lot of music where the tracks have weird tagging information like "[SOMETHING] Actual Track Name" in title and then similarly "[RANDOM] Artist Name" in Artist tag. I'd like to get rid of these irrational tags before searching meta sources.

On top of that, I have music that you wouldn't find on MusicBrainz. Most of my Punjabi music collection can be found on the more mainstream Spotify, Apple Music, Google Music, etc. So I used the Spotify plugin, but when it searches it uses the tag information from the file. So I'll get track results for stuff like "SOMETHING" or "RANDOM", rather than the actual content of the title.

Is there a way to regex replace before sending to spotify/meta sources? So when the queries get sent to spotify, spotify reponds to the actual track/album name, rather than the irrational tag prefix, suffixes, etc.

I tried the [importreplace plugin](https://github.com/edgars-supe/beets-importreplace), but it seems to modify tags later in the process, rather than on the initial "import" search.

Log:
```
> beet -v im [SOMETHING] Actual Track Name - [RANDOM] Artist Name.m4a
...
spotify: Searching Spotify for 'album:[SOMETHING] Album name artist:[RANDOM] Artist Name'
spotify: Found 0 result(s) from Spotify for 'album:[SOMETHING] Album name artist:[RANDOM] Artist Name'
Evaluating 5 candidates.

[SOMETHING] Actual Track Name - [RANDOM] Artist Name.m4a (1 items)
Sending event: import_task_before_choice
Sending event: before_choose_candidate
Finding tags for album "[SOMETHING] Actual Track Name - [RANDOM] Artist Name".
Candidates:
1. Unmatched Something Music Brainz 1
2. Unmatched Random Music Brainz 2
3. Unmatched Music Brainz 3
4. Unmatched Music Brainz 4
5. Unmatched Music Brainz 5
# selection (default 1), Skip, Use as-is, as Tracks, Group albums,
Enter search, enter Id, aBort, eDit, edit Candidates, Print tracks? B
```

PS: If I manually enter the "Enter search" (last prompt) criteria to be "Artist name", "Album name", etc, Spotify, et all will be able to find it.

Thanks.

**Comment by ybnd** [comment]
I'm pretty sure you should be able to work around this as follows:

1. Import without autotagging, keeping the original metadata
    ```beet import -A <import path>```
2. [Modify the the metadata in the database to get rid of unwanted text](https://beets.readthedocs.io/en/stable/reference/cli.html#modify)
3. Reimport from the library, this time with autotagging
     ```beet import -L <query imported tracks/albums>```

This would involve a bit of busywork though, depending on how much files have this issue we may need to find a more clever solution

**Reply by inderpreet99** [reply]
Hi @ybnd, thanks for that suggestion. I tried the `beet import -A` option. Learned that "albuminfo_received" or "trackinfo_received" hooks are not fired, so I can't use the importreplace plugin to programmatically cleanup (i.e. with regexes). It seems modifying manually using the `beet modify` is the only option (as you suggested), but it's busywork and tiresome. Let me know if there's a clever solution.

I realized that I'm actually needing a feature like Formatted modify (https://github.com/beetbox/beets/issues/488), which has been oft requested in the past. Hopefully that associated PR comes through.
